[Elastic Agent] Improve version, restart, enroll CLI commands

blakerouse opened the pull request:
<!-- Type of change
Please label this PR with one of the following labels, depending on the scope of your change:
- Bug
- Enhancement
- Breaking change
- Deprecation
-->

## What does this PR do?

<!-- Mandatory
Explain here the changes you made on the PR. Please explain the WHAT: patterns used, algorithms implemented, design architecture, message processing, etc.
-->

`version` has been changed to show both the version of the running daemon and the version of the executing binary. Along with `--binary-only` argument to skip connecting and reporting the version of the daemon and `--yaml` to print machine parsable version information.

`enroll` has been updated to connect to the running daemon and perform a restart. This is because after enroll if the daemon is already started it needs to be restart to load the new mode of communicating with Fleet.

`restart` is a new command added to just trigger restart of the current running daemon. This is very useful to test re-execution (especially on Windows, because SIGHUP doesn't exist on Windows).

This also includes some fixes that I thought was in my last PR for Windows, except I forgot to commit before merging the PR.

## Why is it important?

<!-- Mandatory
Explain here the WHY, or the rationale/motivation for the changes.
-->

To show the current running daemon version of the Agent, which could be different then the user executing binary. To improve the UX of the enroll command to re-start the daemon once successfully enrolled. To help debugging of re-execution.

## Checklist

<!-- Mandatory
Add a checklist of things that are required to be reviewed in order to have the PR approved

List here all the items you have verified BEFORE sending this PR. Please DO NOT remove any item, striking through those that do not apply. (Just in case, strikethrough uses two tildes. ~~Scratch this.~~)
-->

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- [X] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

<!-- Recommended
Link related issues below. Insert the issue link or reference after the word "Closes" if merging this should automatically close it.

- Closes #123
- Relates #123
- Requires #123
- Superseds #123
-->
- Relates #20142 

## Logs

<!-- Recommended
Paste here output logs discovered while creating this PR, such as stack traces or integration logs, or any other output you consider important to be shared with the team.
-->

### `version`

```
blake@MacBook-Pro elastic-agent % ./elastic_agent version
Daemon: 8.0.0 (build: unknown at 0001-01-01 00:00:00 +0000 UTC)
Binary: 8.0.0 (build: unknown at 0001-01-01 00:00:00 +0000 UTC)
```

```
blake@MacBook-Pro elastic-agent % ./elastic_agent version --yaml
binary:
  version: 8.0.0
  commit: unknown
  build_time: 0001-01-01T00:00:00Z
  snapshot: false
daemon:
  version: 8.0.0
  commit: unknown
  build_time: 0001-01-01T00:00:00Z
  snapshot: false
```

```
blake@MacBook-Pro elastic-agent % ./elastic_agent version --binary-only --yaml
binary:
  version: 8.0.0
  commit: unknown
  build_time: 0001-01-01T00:00:00Z
  snapshot: false
```

### `enroll`

```
blake@MacBook-Pro elastic-agent % ./elastic_agent enroll http://localhost:5061 LVY3Tm9ITUItQ1RKdjBvVU9nN1Q6WkNQQTdsMUlUbWVrOWRsQ3c3QUszdw --insecure --force
The Elastic Agent is currently in BETA and should not be used in production
2020-07-30T13:41:11.926-0400	DEBUG	kibana/client.go:170	Request method: POST, path: /api/ingest_manager/fleet/agents/enroll
Successfully enrolled the Elastic Agent.
Successfully triggered restart on running Elastic Agent.
```

### `restart`

```
blake@MacBook-Pro elastic-agent % ./elastic_agent restart
blake@MacBook-Pro elastic-agent %
```

elasticmachine commented:
Pinging @elastic/ingest-management (Team:Ingest Management)

elasticmachine commented:
## :grey_exclamation: Build Aborted
  
  > There is a new build on-going so the previous on-going builds have been aborted.
  


<!-- BUILD BADGES-->
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-20359/detail/PR-20359/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-20359/detail/PR-20359/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-20359/detail/PR-20359/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-20359/detail/PR-20359/1//artifacts) 

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats

* Build Cause: `[Pull request #20359 opened]`

* Reason: `Aborted from #2`

* Start Time: 2020-07-30T17:43:46.104+0000

* Duration: 6 min 31 sec


* Commit: 85e0d89e0c4ffa46c29740fed651aa35c1805fd9




</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY-->




<!-- LOG OUTPUT-->
### Log output

<details><summary>Expand to view the last 100 lines of log output</summary>
<p>

```
[2020-07-30T17:49:06.806Z] Stage "Elastic Agent Mac OS X" skipped due to earlier failure(s)
[2020-07-30T17:49:06.821Z] Stage "Filebeat oss" skipped due to earlier failure(s)
[2020-07-30T17:49:06.836Z] Stage "Filebeat x-pack" skipped due to earlier failure(s)
[2020-07-30T17:49:06.850Z] Stage "Filebeat Mac OS X" skipped due to earlier failure(s)
[2020-07-30T17:49:06.866Z] Stage "Filebeat x-pack Mac OS X" skipped due to earlier failure(s)
[2020-07-30T17:49:06.882Z] Stage "Filebeat Windows" skipped due to earlier failure(s)
[2020-07-30T17:49:06.897Z] Stage "Filebeat x-pack Windows" skipped due to earlier failure(s)
[2020-07-30T17:49:06.919Z] Stage "Heartbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:06.928Z] Stage "Auditbeat oss Linux" skipped due to earlier failure(s)
[2020-07-30T17:49:06.946Z] Stage "Auditbeat crosscompile" skipped due to earlier failure(s)
[2020-07-30T17:49:06.964Z] Stage "Auditbeat oss Mac OS X" skipped due to earlier failure(s)
[2020-07-30T17:49:06.979Z] Stage "Auditbeat oss Windows" skipped due to earlier failure(s)
[2020-07-30T17:49:06.996Z] Stage "Auditbeat x-pack" skipped due to earlier failure(s)
[2020-07-30T17:49:07.012Z] Stage "Auditbeat x-pack Mac OS X" skipped due to earlier failure(s)
[2020-07-30T17:49:07.028Z] Stage "Auditbeat x-pack Windows" skipped due to earlier failure(s)
[2020-07-30T17:49:07.042Z] Stage "Libbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:07.049Z] Stage "Libbeat x-pack" skipped due to earlier failure(s)
[2020-07-30T17:49:07.066Z] Stage "Metricbeat OSS Unit tests" skipped due to earlier failure(s)
[2020-07-30T17:49:07.082Z] Stage "Metricbeat OSS Go Integration tests" skipped due to earlier failure(s)
[2020-07-30T17:49:07.098Z] Stage "Metricbeat OSS Python Integration tests" skipped due to earlier failure(s)
[2020-07-30T17:49:07.117Z] Stage "Metricbeat x-pack" skipped due to earlier failure(s)
[2020-07-30T17:49:07.127Z] Stage "Metricbeat crosscompile" skipped due to earlier failure(s)
[2020-07-30T17:49:07.144Z] Stage "Metricbeat Mac OS X" skipped due to earlier failure(s)
[2020-07-30T17:49:07.159Z] Stage "Metricbeat x-pack Mac OS X" skipped due to earlier failure(s)
[2020-07-30T17:49:07.176Z] Stage "Metricbeat Windows" skipped due to earlier failure(s)
[2020-07-30T17:49:07.191Z] Stage "Metricbeat x-pack Windows" skipped due to earlier failure(s)
[2020-07-30T17:49:07.208Z] Stage "Packetbeat OSS" skipped due to earlier failure(s)
[2020-07-30T17:49:07.216Z] Stage "dockerlogbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:07.224Z] Stage "Winlogbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:07.231Z] Stage "Winlogbeat Windows x-pack" skipped due to earlier failure(s)
[2020-07-30T17:49:07.246Z] Stage "Functionbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:07.253Z] Stage "Journalbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:07.261Z] Stage "Generators" skipped due to earlier failure(s)
[2020-07-30T17:49:07.270Z] Stage "Kubernetes" skipped due to earlier failure(s)
[2020-07-30T17:49:07.657Z] Stage "Heartbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:07.671Z] Stage "Libbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:07.683Z] Stage "Metricbeat x-pack" skipped due to earlier failure(s)
[2020-07-30T17:49:07.697Z] Stage "Packetbeat OSS" skipped due to earlier failure(s)
[2020-07-30T17:49:07.711Z] Stage "dockerlogbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:07.724Z] Stage "Winlogbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:07.739Z] Stage "Functionbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:07.755Z] Stage "Journalbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:07.770Z] Stage "Generators" skipped due to earlier failure(s)
[2020-07-30T17:49:10.051Z] Failed in branch Elastic Agent x-pack
[2020-07-30T17:49:10.065Z] Failed in branch Elastic Agent x-pack Windows
[2020-07-30T17:49:10.078Z] Failed in branch Elastic Agent Mac OS X
[2020-07-30T17:49:10.089Z] Failed in branch Filebeat oss
[2020-07-30T17:49:10.100Z] Failed in branch Filebeat x-pack
[2020-07-30T17:49:10.114Z] Failed in branch Filebeat Mac OS X
[2020-07-30T17:49:10.126Z] Failed in branch Filebeat x-pack Mac OS X
[2020-07-30T17:49:10.137Z] Failed in branch Filebeat Windows
[2020-07-30T17:49:10.149Z] Failed in branch Filebeat x-pack Windows
[2020-07-30T17:49:10.160Z] Failed in branch Auditbeat oss Linux
[2020-07-30T17:49:10.172Z] Failed in branch Auditbeat crosscompile
[2020-07-30T17:49:10.183Z] Failed in branch Auditbeat oss Mac OS X
[2020-07-30T17:49:10.195Z] Failed in branch Auditbeat oss Windows
[2020-07-30T17:49:10.206Z] Failed in branch Auditbeat x-pack
[2020-07-30T17:49:10.220Z] Failed in branch Auditbeat x-pack Mac OS X
[2020-07-30T17:49:10.232Z] Failed in branch Auditbeat x-pack Windows
[2020-07-30T17:49:10.245Z] Failed in branch Libbeat x-pack
[2020-07-30T17:49:10.257Z] Failed in branch Metricbeat OSS Unit tests
[2020-07-30T17:49:10.269Z] Failed in branch Metricbeat OSS Go Integration tests
[2020-07-30T17:49:10.280Z] Failed in branch Metricbeat OSS Python Integration tests
[2020-07-30T17:49:10.292Z] Failed in branch Metricbeat crosscompile
[2020-07-30T17:49:10.305Z] Failed in branch Metricbeat Mac OS X
[2020-07-30T17:49:10.316Z] Failed in branch Metricbeat x-pack Mac OS X
[2020-07-30T17:49:10.331Z] Failed in branch Metricbeat Windows
[2020-07-30T17:49:10.342Z] Failed in branch Metricbeat x-pack Windows
[2020-07-30T17:49:10.352Z] Failed in branch Winlogbeat Windows x-pack
[2020-07-30T17:49:10.363Z] Failed in branch Kubernetes
[2020-07-30T17:49:11.020Z] Stage "Heartbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:11.034Z] Stage "Libbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:11.047Z] Stage "Metricbeat x-pack" skipped due to earlier failure(s)
[2020-07-30T17:49:11.058Z] Stage "Packetbeat OSS" skipped due to earlier failure(s)
[2020-07-30T17:49:11.072Z] Stage "Winlogbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:11.089Z] Stage "Functionbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:11.102Z] Stage "Generators" skipped due to earlier failure(s)
[2020-07-30T17:49:11.558Z] Failed in branch dockerlogbeat
[2020-07-30T17:49:11.571Z] Failed in branch Journalbeat
[2020-07-30T17:49:12.170Z] Stage "Heartbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:12.183Z] Stage "Libbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:12.195Z] Stage "Packetbeat OSS" skipped due to earlier failure(s)
[2020-07-30T17:49:12.207Z] Stage "Functionbeat" skipped due to earlier failure(s)
[2020-07-30T17:49:12.220Z] Stage "Generators" skipped due to earlier failure(s)
[2020-07-30T17:49:12.348Z] Failed in branch Metricbeat x-pack
[2020-07-30T17:49:12.359Z] Failed in branch Winlogbeat
[2020-07-30T17:49:12.920Z] Failed in branch Heartbeat
[2020-07-30T17:49:12.933Z] Failed in branch Libbeat
[2020-07-30T17:49:12.946Z] Failed in branch Packetbeat OSS
[2020-07-30T17:49:12.958Z] Failed in branch Functionbeat
[2020-07-30T17:49:12.959Z] Stage "Generators" skipped due to earlier failure(s)
[2020-07-30T17:49:13.201Z] Failed in branch Generators
[2020-07-30T17:49:15.827Z] Running on Jenkins in /var/lib/jenkins/workspace/Beats_beats_PR-20359
[2020-07-30T17:49:16.153Z] [INFO] getVaultSecret: Getting secrets
[2020-07-30T17:49:16.241Z] Masking supported pattern matches of $VAULT_ADDR or $VAULT_ROLE_ID or $VAULT_SECRET_ID
[2020-07-30T17:49:17.368Z] + chmod 755 generate-build-data.sh
[2020-07-30T17:49:17.369Z] + ./generate-build-data.sh https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-20359/ https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-20359/runs/1 ABORTED 330992
[2020-07-30T17:49:17.369Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-20359/runs/1/steps/?limit=10000 -o steps-info.json
[2020-07-30T17:49:17.619Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-20359/runs/1/tests/?status=FAILED -o tests-errors.json
[2020-07-30T17:49:17.619Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-20359/runs/1/log/ -o pipeline-log.txt

```

</p>
</details>



<!--COMMENT_GENERATED_WITH_ID_comment.id-->

michalpristas commented on x-pack/elastic-agent/pkg/agent/application/reexec/manager.go:
```diff
@@ -30,14 +30,19 @@ type ExecManager interface {
 
+// Manager returns the global reexec manager.
+func Manager() ExecManager {
+	return execSingleton
```
This does not look like a good idea why not having just one method with Once-initialization

blakerouse replied:
I would like to, but that deep into the code, I didn't want to have to pass in the logger and the path to the executable. I wanted that to be set at startup and later usages of the manager, can get the singleton.

michalpristas commented on x-pack/elastic-agent/pkg/agent/cmd/enroll.go:
```diff
@@ -144,7 +148,25 @@ func enroll(streams *cli.IOStreams, cmd *cobra.Command, flags *globalFlags, args
+	daemon := client.New()
+	err = daemon.Start(context.Background())
+	if err == nil {
+		defer daemon.Stop()
```
why are you stopping after restart is done?

blakerouse replied:
`Stop` just disconnects the client from the socket, it does not stop the running daemon.

michalpristas replied:
this all makes sense then :-)

michalpristas commented on x-pack/elastic-agent/pkg/agent/control/server/listener.go:
```diff
@@ -17,6 +17,9 @@ import (
 func createListener() (net.Listener, error) {
 	path := strings.TrimPrefix(control.Address(), "unix://")
+	if _, err := os.Stat(path); !os.IsNotExist(err) {
+		os.Remove(path)
```
can we check for errors here, if file exists but you're revoked access to it (too many fd, process reading/writing to it already whatever...) this will fail and we wont know

blakerouse replied:
I will log it, but the error of not creating the listener socket will still be the really reported error.

michalpristas commented on x-pack/elastic-agent/pkg/agent/control/server/listener.go:
```diff
@@ -36,3 +39,8 @@ func createListener() (net.Listener, error) {
+
+func cleanupListener() {
+	path := strings.TrimPrefix(control.Address(), "unix://")
+	os.Remove(path)
```
same here, at least log error returned from remove, will make troubleshooting easier

blakerouse replied:
Same as above, will log it.

michalpristas commented on x-pack/elastic-agent/pkg/basecmd/restart/cmd.go:
```diff
@@ -0,0 +1,36 @@
+			c := client.New()
+			err := c.Start(context.Background())
+			if err != nil {
+				return errors.New(err, "Failed talking to running daemon")
```
connecting to?

blakerouse replied:
Want me to add the path to the unix socket or the npipe?

michalpristas replied:
no i meant `talking to` phrase feels weird to me

michalpristas commented on x-pack/elastic-agent/pkg/basecmd/restart/cmd.go:
```diff
@@ -0,0 +1,36 @@
+			if err != nil {
+				return errors.New(err, "Failed talking to running daemon")
+			}
+			defer c.Stop()
```
why are we stopping after restart?

blakerouse replied:
As above, this just disconnects from the running daemon.

michalpristas commented on x-pack/elastic-agent/pkg/basecmd/version/cmd.go:
```diff
@@ -5,32 +5,95 @@
+				c := client.New()
+				daemonError = c.Start(context.Background())
+				if daemonError == nil {
+					defer c.Stop()
```
i think what stop does and what i think it does are two different things

blakerouse replied:
Same as above.

blakerouse commented on x-pack/elastic-agent/CHANGELOG.asciidoc:
```diff
@@ -99,3 +99,6 @@
 - Add --staging option to enroll command {pull}20026[20026]
 - Add `event.dataset` to all events {pull}20076[20076]
 - Prepare packaging for endpoint and asc files {pull}20186[20186]
+- Improved version CLI {pull}20359[20359]
```
I thought it was easier to read being 3 different things, but if you prefer it to be 1 line, I can do that.

blakerouse commented:
@michalpristas Thanks for the +1.

I have updated the client from `Start` and `Stop` to `Connect` and `Disconnect` to make that more clear. I changed `reexec` module to not be a singleton anymore, which makes it easier to use and cleaner (as I generally don't like singletons, idk what I was thinking originally 😄  ). I also added the logging, you asked for.

michalpristas commented:
@blakerouse thanks for rename, it makes it much more readable

elasticmachine commented:
## :grey_exclamation: Build Aborted
  
  > There is a new build on-going so the previous on-going builds have been aborted.
  


<!-- BUILD BADGES-->
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-20359/detail/PR-20359/3//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-20359/detail/PR-20359/3//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-20359/detail/PR-20359/3//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-20359/detail/PR-20359/3//artifacts) 

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats

* Build Cause: `[Pull request #20359 updated]`

* Reason: `Aborted from #4`

* Start Time: 2020-08-03T14:28:51.940+0000

* Duration: 122 min 46 sec


* Commit: a3342535e24257f38f2c4353e81d9103b61688d5




</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY-->




<!-- LOG OUTPUT-->
### Log output

<details><summary>Expand to view the last 100 lines of log output</summary>
<p>

```
[2020-08-03T16:28:47.038Z] ERROR 
[2020-08-03T16:28:47.087Z] Running in C:\Users\jenkins\workspace\Beats_beats_PR-20359\src\github.com\elastic\beats\x-pack\elastic-agent
[2020-08-03T16:28:47.396Z] 
[2020-08-03T16:28:47.396Z] C:\Users\jenkins\workspace\Beats_beats_PR-20359\src\github.com\elastic\beats\x-pack\elastic-agent>mage  build unitTest 
[2020-08-03T16:29:34.134Z] go version go1.14.4 windows/amd64
[2020-08-03T16:29:34.134Z] set GO111MODULE=
[2020-08-03T16:29:34.134Z] set GOARCH=amd64
[2020-08-03T16:29:34.134Z] set GOBIN=
[2020-08-03T16:29:34.134Z] set GOCACHE=C:\Users\jenkins.PACKER-5D9FFF00\AppData\Local\go-build
[2020-08-03T16:29:34.134Z] set GOENV=C:\Users\jenkins.PACKER-5D9FFF00\AppData\Roaming\go\env
[2020-08-03T16:29:34.134Z] set GOEXE=.exe
[2020-08-03T16:29:34.134Z] set GOFLAGS=
[2020-08-03T16:29:34.134Z] set GOHOSTARCH=amd64
[2020-08-03T16:29:34.134Z] set GOHOSTOS=windows
[2020-08-03T16:29:34.134Z] set GOINSECURE=
[2020-08-03T16:29:34.134Z] set GONOPROXY=
[2020-08-03T16:29:34.134Z] set GONOSUMDB=
[2020-08-03T16:29:34.134Z] set GOOS=windows
[2020-08-03T16:29:34.134Z] set GOPATH=C:\Users\jenkins\workspace\Beats_beats_PR-20359
[2020-08-03T16:29:34.134Z] set GOPRIVATE=
[2020-08-03T16:29:34.134Z] set GOPROXY=https://proxy.golang.org,direct
[2020-08-03T16:29:34.134Z] set GOROOT=C:\Users\jenkins.PACKER-5D9FFF00\.gvm\versions\go1.14.4.windows.amd64
[2020-08-03T16:29:34.134Z] set GOSUMDB=sum.golang.org
[2020-08-03T16:29:34.134Z] set GOTMPDIR=
[2020-08-03T16:29:34.134Z] set GOTOOLDIR=C:\Users\jenkins.PACKER-5D9FFF00\.gvm\versions\go1.14.4.windows.amd64\pkg\tool\windows_amd64
[2020-08-03T16:29:34.134Z] set GCCGO=gccgo
[2020-08-03T16:29:34.134Z] set AR=ar
[2020-08-03T16:29:34.134Z] set CC=gcc
[2020-08-03T16:29:34.134Z] set CXX=g++
[2020-08-03T16:29:34.134Z] set CGO_ENABLED=1
[2020-08-03T16:29:34.134Z] set GOMOD=C:\Users\jenkins\workspace\Beats_beats_PR-20359\src\github.com\elastic\beats\go.mod
[2020-08-03T16:29:34.134Z] set CGO_CFLAGS=-g -O2
[2020-08-03T16:29:34.134Z] set CGO_CPPFLAGS=
[2020-08-03T16:29:34.134Z] set CGO_CXXFLAGS=-g -O2
[2020-08-03T16:29:34.134Z] set CGO_FFLAGS=-g -O2
[2020-08-03T16:29:34.134Z] set CGO_LDFLAGS=-g -O2
[2020-08-03T16:29:34.134Z] set PKG_CONFIG=pkg-config
[2020-08-03T16:29:34.134Z] set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0 -fdebug-prefix-map=C:\Users\JENKIN~1.PAC\AppData\Local\Temp\go-build001598794=/tmp/go-build -gno-record-gcc-switches
[2020-08-03T16:29:34.134Z] >> build: Building elastic-agent
[2020-08-03T16:30:12.843Z] === RUN   TestSystem
[2020-08-03T16:30:12.843Z] --- PASS: TestSystem (0.00s)
[2020-08-03T16:30:12.843Z] PASS
[2020-08-03T16:30:12.843Z] coverage: 0.0% of statements
[2020-08-03T16:30:12.843Z] ok  	github.com/elastic/beats/v7/x-pack/elastic-agent	0.172s	coverage: 0.0% of statements
[2020-08-03T16:30:13.216Z] 
[2020-08-03T16:30:13.216Z] C:\Users\jenkins\workspace\Beats_beats_PR-20359\src\github.com\elastic\beats>python .ci/scripts/pre_archive_test.py 
[2020-08-03T16:30:14.151Z] Copy .\x-pack\elastic-agent\build into build\x-pack\elastic-agent\build
[2020-08-03T16:30:14.158Z] Running in C:\Users\jenkins\workspace\Beats_beats_PR-20359\src\github.com\elastic\beats\build
[2020-08-03T16:30:14.182Z] Recording test results
[2020-08-03T16:30:14.624Z] None of the test reports contained any result
[2020-08-03T16:30:14.771Z] Stashed 0 file(s)
[2020-08-03T16:30:14.780Z] Archiving artifacts
[2020-08-03T16:30:14.839Z] ‘**\build\TEST*.out’ doesn’t match anything: ‘**’ exists but not ‘**\build\TEST*.out’
[2020-08-03T16:30:14.839Z] No artifacts found that match the file pattern "**\build\TEST*.out". Configuration error?
[2020-08-03T16:30:15.249Z] [INFO] system-tests=''. If no empty then let's create a tarball
[2020-08-03T16:30:16.983Z] Running in /var/lib/jenkins/workspace/Beats_beats_PR-20359/src/github.com/elastic/beats
[2020-08-03T16:30:17.296Z] + find . -type f -name TEST*.xml -path */build/* -delete
[2020-08-03T16:30:17.309Z] Running in /var/lib/jenkins/workspace/Beats_beats_PR-20359/src/github.com/elastic/beats/Lint
[2020-08-03T16:30:17.410Z] Running in /var/lib/jenkins/workspace/Beats_beats_PR-20359/src/github.com/elastic/beats/Elastic-Agent-Mac-OS-X
[2020-08-03T16:30:17.488Z] Running in /var/lib/jenkins/workspace/Beats_beats_PR-20359/src/github.com/elastic/beats/Elastic-Agent-x-pack
[2020-08-03T16:30:17.572Z] Running in /var/lib/jenkins/workspace/Beats_beats_PR-20359/src/github.com/elastic/beats/Elastic-Agent-x-pack-Windows
[2020-08-03T16:30:17.941Z] + cat
[2020-08-03T16:30:17.941Z] + /usr/local/bin/runbld ./runbld-script
[2020-08-03T16:30:17.941Z] Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
[2020-08-03T16:30:24.533Z] runbld>>> runbld started
[2020-08-03T16:30:24.533Z] runbld>>> 1.6.12/f45d832f2ba0aa2722ab4ec1fda8ad140f027f8b
[2020-08-03T16:30:26.451Z] runbld>>> The following profiles matched the job 'Beats/beats/PR-20359' in order of occurrence in the config (last value wins).
[2020-08-03T16:30:27.836Z] runbld>>> Debug logging enabled.
[2020-08-03T16:30:27.836Z] runbld>>> Storing result
[2020-08-03T16:30:27.836Z] runbld>>> Store result: created {:total 2, :successful 2, :failed 0} 1
[2020-08-03T16:30:27.836Z] runbld>>> BUILD: https://c150076387b5421f9154dfbf536e5c60.us-west1.gcp.cloud.es.io:9243/build-1587637540455/t/20200803163027-C3CA6BE4
[2020-08-03T16:30:27.836Z] runbld>>> Adding system facts.
[2020-08-03T16:30:29.229Z] runbld>>> Adding vcs info for the latest commit:  d7e29fd51c36d7d7293c1de5d5a9ff95ea4e2deb
[2020-08-03T16:30:29.229Z] runbld>>> >>>>>>>>>>>> SCRIPT EXECUTION BEGIN >>>>>>>>>>>>
[2020-08-03T16:30:29.229Z] runbld>>> Adding /usr/lib/jvm/java-8-openjdk-amd64/bin to the path.
[2020-08-03T16:30:29.229Z] Processing JUnit reports with runbld...
[2020-08-03T16:30:29.229Z] + echo 'Processing JUnit reports with runbld...'
[2020-08-03T16:30:29.490Z] runbld>>> <<<<<<<<<<<< SCRIPT EXECUTION END <<<<<<<<<<<<
[2020-08-03T16:30:29.490Z] runbld>>> DURATION: 27ms
[2020-08-03T16:30:29.490Z] runbld>>> STDOUT: 40 bytes
[2020-08-03T16:30:29.490Z] runbld>>> STDERR: 49 bytes
[2020-08-03T16:30:29.490Z] runbld>>> WRAPPED PROCESS: SUCCESS (0)
[2020-08-03T16:30:29.490Z] runbld>>> Searching for build metadata in /var/lib/jenkins/workspace/Beats_beats_PR-20359/src/github.com/elastic/beats
[2020-08-03T16:30:30.434Z] runbld>>> Storing build metadata: 
[2020-08-03T16:30:30.434Z] runbld>>> Adding test report.
[2020-08-03T16:30:30.434Z] runbld>>> Searching for junit test output files with the pattern: TEST-.*\.xml$ in: /var/lib/jenkins/workspace/Beats_beats_PR-20359/src/github.com/elastic/beats
[2020-08-03T16:30:31.377Z] runbld>>> Found 0 test output files
[2020-08-03T16:30:31.377Z] runbld>>> Test output logs contained: Errors: 0 Failures: 0 Tests: 0 Skipped: 0
[2020-08-03T16:30:31.377Z] runbld>>> Storing result
[2020-08-03T16:30:31.638Z] runbld>>> Store result: updated {:total 2, :successful 2, :failed 0} 2
[2020-08-03T16:30:31.638Z] runbld>>> BUILD: https://c150076387b5421f9154dfbf536e5c60.us-west1.gcp.cloud.es.io:9243/build-1587637540455/t/20200803163027-C3CA6BE4
[2020-08-03T16:30:31.899Z] runbld>>> Email notification disabled by environment variable.
[2020-08-03T16:30:31.899Z] runbld>>> Slack notification disabled by environment variable.
[2020-08-03T16:30:37.419Z] Running on Jenkins in /var/lib/jenkins/workspace/Beats_beats_PR-20359
[2020-08-03T16:30:37.515Z] [INFO] getVaultSecret: Getting secrets
[2020-08-03T16:30:37.581Z] Masking supported pattern matches of $VAULT_ADDR or $VAULT_ROLE_ID or $VAULT_SECRET_ID
[2020-08-03T16:30:38.310Z] + chmod 755 generate-build-data.sh
[2020-08-03T16:30:38.310Z] + ./generate-build-data.sh https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-20359/ https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-20359/runs/3 ABORTED 7306110
[2020-08-03T16:30:38.310Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-20359/runs/3/steps/?limit=10000 -o steps-info.json
[2020-08-03T16:30:38.560Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-20359/runs/3/tests/?status=FAILED -o tests-errors.json

```

</p>
</details>



<!--COMMENT_GENERATED_WITH_ID_comment.id-->
